diagnosisApplication/src/main/use_case/diagnosis/DiagnosisOutputData.java

```diff
@@ -1,18 +1,49 @@
 package main.use_case.diagnosis;
 
 import java.util.HashMap;
-import java.util.List;
 
 public class DiagnosisOutputData {
 
-    private final List<Integer> checkedSymptoms;
+    private final HashMap<String, Object> diagnosis1;
+    private final HashMap<String, Object> diagnosis2;
+    private final HashMap<String, Object> diagnosis3;
 
-    public DiagnosisOutputData(List<Integer> checkedSymptoms) { this.checkedSymptoms = checkedSymptoms; }
+    public DiagnosisOutputData(HashMap<String, Object> diagnosis1,
+                               HashMap<String, Object> diagnosis2,
+                               HashMap<String, Object> diagnosis3) {
+        this.diagnosis1 = diagnosis1;
+        this.diagnosis2 = diagnosis2;
+        this.diagnosis3 = diagnosis3;
+    }
+
+    public DiagnosisOutputData(HashMap<String, Object> diagnosis1,
+                               HashMap<String, Object> diagnosis2) {
+        this.diagnosis1 = diagnosis1;
+        this.diagnosis2 = diagnosis2;
+        this.diagnosis3 = null;
+    }
+
+    public DiagnosisOutputData(HashMap<String, Object> diagnosis1) {
+        this.diagnosis1 = diagnosis1;
+        this.diagnosis2 = null;
+        this.diagnosis3 = null;
+    }
+
+    public DiagnosisOutputData() {
+        this.diagnosis1 = null;
+        this.diagnosis2 = null;
+        this.diagnosis3 = null;
+    }
+
+    public HashMap<String, Object> getDiagnosis1() {
+        return diagnosis1;
+    }
```

Comment:
Setup diagnosis outputs using overloading for API outputs with different amounts of diseases to cover different output cases.